algorithms/src/encryption/ecies_poseidon.rs

```diff
@@ -226,121 +226,141 @@ where
     ///     ciphertext := to_bytes_le![C_1, ..., C_n], where C_i := R_i + M_i, and R_i := H_i(G^ar)
     /// ```
     ///
-    fn encrypt(&self, symmetric_key: &Self::SymmetricKey, message: &[u8]) -> Result<Vec<u8>, EncryptionError> {
+    fn encrypt(
+        &self,
+        symmetric_key: &Self::SymmetricKey,
+        message: &[Vec<u8>],
```

Comment:
would it be possible to change `&[Vec<u8>]` to `&[&[u8]]`? we don't really need those elements to be owned vectors; otherwise, something like `&[T]` where `T: AsRef<[u8]>` would work too.